lib/kafo/scenario_manager.rb

```diff
@@ -78,12 +78,19 @@ def configured?
 
     def scenario_from_args(arg_name='--scenario|-S')
       # try scenario provided in the args via -S or --scenario
-      parsed = ARGV.join(" ").match(/(#{arg_name})(\s+|[=]?)(\S+)/)
-      if parsed
-        scenario_file = File.join(config_dir, "#{parsed[3]}.yaml")
-        return scenario_file if File.exist?(scenario_file)
-        fail_now("Scenario (#{scenario_file}) was not found, can not continue", :unset_scenario)
+      ARGV.each_with_index do |arg, index|
+        parsed = arg.match(/^(#{arg_name})($|=(?<scenario>\S+))/)
+        if parsed
+          scenario = parsed[:scenario] || ARGV[index + 1]
```

Comment:
In the example I did with `ARGV = ["--scenario", "katello", "--cert", "A-ST.pem"]`, I found that the values of parsed were [nil, nil, #<MatchData "--scenario" scenario:nil>, nil] respectively.

Could you explain what `parsed[:scenario]` is doing here?

Reply:
The part `(?<scenario>\S+)` is a [named capture group](https://ruby-doc.org/core-2.7.1/Regexp.html#class-Regexp-label-Capturing). If you don't do that, you need to refer to it by number. Since a number can change is hard to follow, I chose a named group.

In the case of `['--scenario', 'katello']` there is no match so it's nil. That's why I do a look forward (`ARGV[index + 1]`) to get the next argument.

Reply:
Great, thanks for the reference